moto/securityhub/models.py

```diff
@@ -0,0 +1,118 @@
+"""SecurityHubBackend class with methods for supported APIs."""
+
+from typing import Any, Dict, List, Optional, Tuple
+
+from moto.core.base_backend import BackendDict, BaseBackend
+from moto.core.common_models import BaseModel
+from moto.securityhub.exceptions import InvalidInputException
+from moto.utilities.paginator import Paginator
+
+
+class Finding(BaseModel):
+    def __init__(self, finding_id: str, finding_data: Dict[str, Any]):
+        self.id = finding_id
+        self.data = finding_data
+
+    def as_dict(self) -> Dict[str, Any]:
+        return self.data
+
+
+class SecurityHubBackend(BaseBackend):
+    """Implementation of SecurityHub APIs."""
+
+    def __init__(self, region_name: str, account_id: str):
+        super().__init__(region_name, account_id)
+        self.findings: List[Finding] = []
+
+    def get_findings(
+        self,
+        filters: Optional[Dict[str, Any]] = None,
+        sort_criteria: Optional[List[Dict[str, str]]] = None,
+        next_token: Optional[str] = None,
+        max_results: Optional[int] = None,
+    ) -> Dict[str, Any]:
+        findings = self.findings
+
+        # Max Results Parameter
+        if max_results is not None:
+            try:
+                max_results = int(max_results)
+                if max_results < 1 or max_results > 100:
+                    raise InvalidInputException(
+                        op="GetFindings",
+                        msg="MaxResults must be a number between 1 and 100",
+                    )
+            except ValueError:
+                raise InvalidInputException(
+                    op="GetFindings", msg="MaxResults must be a number greater than 0"
+                )
+
+        paginator = Paginator(
```

Comment:
It looks like we can just use the `@paginate`-decorator on the method here, that should simplify the implementation a bit.

Please see the docs here: https://docs.getmoto.org/en/latest/docs/contributing/development_tips/utilities.html#paginator